staging/src/k8s.io/apimachinery/pkg/api/resource/quantity_test.go

```diff
@@ -215,6 +215,48 @@ func TestParseQuantityString(t *testing.T) {
 	}
 }
 
+func TestQuantityToInt64(t *testing.T) {
+	table := []struct {
+		name                 string
+		v                    string
+		notOverflow          bool
+		expectedDecimalValue int64
+	}{
+		{"decimalSI to int64 not overflow",
+			"1000000000G",
+			true,
+			1000000000000000000,
+		},
+		{"binarySI to int64 not overflow",
+			"100000000Gi",
+			true,
+			107374182400000000,
+		},
+		{"decimalSI to int64 overflow",
+			// MaxInt64
+			"9223372036854775807",
+			false,
+			0,
```

Comment:
Can we check the inf.Dec value when `notOverflow` is false? In this case the expected value is 9223372036854775807. 

LGTM otherwise.

Reply:
> Can we check the inf.Dec value when `notOverflow` is false? In this case the expected value is 9223372036854775807.
> 
> LGTM otherwise.

I think zero value is better when  inf.Dec value is larger than MaxInt64